R/utils-github.R

```diff
@@ -826,12 +855,173 @@ cfg_upstream_but_origin_is_not_fork <- function(cfg) {
     list(
       type = "upstream_but_origin_is_not_fork",
       pr_ready = FALSE,
-      desc = glue("
-        Both {ui_value('origin')} and {ui_value('upstream')} are GitHub \\
-        remotes, but {ui_value('origin')} is not a fork and, in particular, \\
-        is not a fork of {ui_value('upstream')}.
-
-        {read_more()}")
+      desc = c(
+        "!" = ui_pre_glue("
+                Both {.val origin} and {.val upstream} are GitHub remotes, but
+                {.val origin} is not a fork and, in particular, is not a fork of
+                {.val upstream}."),
+        read_more()
+      )
     )
   )
 }
+
+# construct instances of `github_remote_config` for dev/testing purposes--------
```

Comment:
The github configuration functionality was tough to work on, because the input (a `remote_github_configuration`) is formed by inspecting a local git repo, its remotes, and calling the GitHub API. I can now make a fake instance of each important configuration, which has been very productive for development and testing.